File: java/Day1.java
public class Day1 extends Day {
    @Override
    int part1(String[] puzzleInput) {
        int counter = 0;
        int[] numbers = ToInt(puzzleInput);

        for (int i = 1; i < numbers.length; i++) {
            if (numbers[i] > numbers[i - 1]) {
                counter++;
            }    
        }

        return counter;
    }

    @Override
    int part2(String[] puzzleInput) {
        int counter = 0;
        int[] numbers = ToInt(puzzleInput);

        for (int i = 0; i < numbers.length - 3; i++) {
            int overlap = numbers[i + 1] + numbers[i + 2];
            int current = numbers[i] + overlap;
            int next = overlap + numbers[i + 3];

            if (next > current) {
                counter++;
            }    
        }

        return counter;
    }

    public static void main(String[] args) {
        new Day1().run(1);
    }
}

File: typescript/Day1.ts
import { BaseDay } from "./BaseDay";

class Day1 extends BaseDay {
    part1(puzzleInput: string[]): string {
        let increaseCounter = 0;
        let puzzleInputAsNumber = this.toInt(puzzleInput);
        let lastInput: number = puzzleInputAsNumber[0];

        puzzleInputAsNumber.forEach(function (input) {
            if (input > lastInput) {
                increaseCounter++;
            }
            lastInput = input;
        })

        return increaseCounter.toString();
    }

    part2(puzzleInput: string[]): string {
        let increaseCounter = 0;
        let puzzleInputAsNumber = this.toInt(puzzleInput);

        for (let i = 0; i < puzzleInputAsNumber.length - 3; i++) {
            let overlap = puzzleInputAsNumber[i + 1] + puzzleInputAsNumber[i + 2];
            let current = puzzleInputAsNumber[i] + overlap;
            let next = overlap + puzzleInputAsNumber[i + 3];

            if (next > current) {
                increaseCounter++;
            }
        }

        return increaseCounter.toString();
    }
}

new Day1().run(1);